### DISCUSSION TITLE
Proxy Providers with JWT Authentication: How to bypass login and fetch a token?

### DISCUSSION THREAD

**Discussion by moutasem1989** [goauthentik/authentik#11764] [Q&A]
The Basic HTTP Authentication comes really handy to bypass a second login for Proxy Providers that use it but cannot disable it: like Kasm Standalone Desktop and Cockpit. But most services use JWT Authentication with no support for OIDC or SSO; like NGINX Proxy Manager.
I assume that requesting a fresh token can be something like this in python:

```
import requests
import json

# Define the URL and the headers
url = 'http://localhost:81/api/tokens'
headers = {
    'Content-Type': 'application/json; charset=UTF-8'
}

# Define the payload with your email and secret
payload = {
    'identity': '[Your Email]',
    'secret': '[Your Secret]'
}

response = requests.post(url, headers=headers, data=json.dumps(payload))

return response.passing
```
Sadly the only [documentation](https://docs.goauthentik.io/docs/add-secure-apps/providers/oauth2/client_credentials) I found is rather unclear. Any help is greatly appreciated!

**Comment by skjdghsdjgsdj** [comment]
Adding my support to see this resolved: I have an [endpoint](https://github.com/gthole/drink-stash?tab=readme-ov-file) that doesn't integrate with OAuth/OpenID/etc. The service does accept an `Authorization: JWT ...` header and I know the secret key it uses for signing its JWTs.

I imagine it's possible with Python code in Authentik to generate the JWT on the fly for proxy authentication, including pulling in OAuth attributes like the current Authentik user's email to fit into the JWT, but I don't see how to do it.
